scarlet/source.py

```diff
@@ -53,13 +53,23 @@ def get_model(self):
                 models[k] = _model
         return np.sum(models, axis=0)
 
+    def get_flux(self):
+        """Get flux in every band
+        """
+        return np.sum([c.get_flux() for c in self.components], axis=0)
+
     def update_center(self):
         """Center update to set all component centers to flux-weighted mean position.
 
         NOTE: Only components with `shift_center > 0` will be moved.
         """
         if len(self.components) > 1:
-            _flux = np.array([c.morph.sum() for c in self.components])
+            # TODO: find optimal centering Weights
+            # With flux weighting, larger components often have large weights
+            # even if they are shallow and thus hard to center ...
+            # _flux = np.array([c.get_flux().sum() for c in self.components])
+            #  ... use flat weights instead
+            _flux = np.array([1 for c in self.components])
```

Comment:
I agree that the older method did not necessarily have the desired affect. Perhaps if we just use a small window (3px x 3 px) we can weight based on the _central_ flux for each component. This should give more weight to components with more flux concentrated at the center, which should be easier to fit. Including the new parameter `center_window` that can either be an array or window size initialized as follows:
```
if np.isscalar(center_window):
    center_window = np.ones((center_window, center_window), dtype=float)
```

This method then becomes:
```
_flux = np.array([np.sum(c.morph[cy-ry:cy+ry+1, cx-rx:cx+rx+1] * self.center_window) for c in self.components])
```
where cy,cx is the location of the central pixel in the morphology and ry, rx is the size of the window.

This allows the user to pass an array like `center_window=np.array([[0,1,0],[1,1,1],[0,1,0]])` to choose the exact window used.

If you agree I can open an issue to make this change.

Reply:
My reply refers to the normalization issue. It's actually correct. There's a `component.normalize` call in *every* source type, which takes care of the different normalizations. The problem is in the `PointSource` initialization only.